repos/system_upgrade/common/actors/missinggpgkeysinhibitor/libraries/missinggpgkey.py

```diff
@@ -0,0 +1,195 @@
+from leapp.libraries.stdlib import api, run
+from leapp.models import InstalledRPM, TMPTargetRepositoriesFacts, UsedTargetRepositories
+from leapp.utils.deprecation import suppress_deprecation
+
+
+# TODO what to do to make sure gnupg2 is installed on the source OS ?
+def _gpg_show_keys(key_path):
+    """
+    Run gpg --show-keys to verify the key exists, is readable and we can read its fingerprint
+    """
+    try:
+        cmd = ['gpg2']
+        # RHEL7 gnupg requires different switches to get the same output
+        if get_source_major_version() == '7':
+            cmd.append('--with-fingerprint')
+        else:
+            cmd.append('--show-keys')
+        cmd += ['--with-colons', key_path]
+        return run(cmd, split=True, checked=False)
+    except OSError as err:
+        error = 'Failed to read fingerprint from GPG key {}: {}'.format(key_path, str(err))
+        api.current_logger().error(error)
+        return []
+
+
+def _parse_fp_from_gpg(output):
+    """
+    Parse the output of gpg --show-keys --with-colons.
+
+    Returns 8 characters fingerprint or none if some error occurred. Either the file
+    does not exists, its not readable or does not contain any gpg key.
+    """
+    if 'stdout' not in output or len(output['stdout']) < 1:
+        return None
+
+    if "can't open" in output['stderr']:
+        return None
+
+    if "no valid OpenPGP data found." in output['stdout']:
+        return None
+
+    # we are interested in the first line of the output.
+    # the colons are used for separating the fields in output like this
+    # pub:-:4096:1:999F7CBF38AB71F4:1612983048:::-:::escESC::::::23::0:
+    #              ^--------------^ this is the fingerprint we need
+    #                      ^------^ but RPM version is just the last 8 chars lowercase
+    parts = output['stdout'][0].split(':')
+    if len(parts) >= 4 and len(parts[4]) == 16:
+        return parts[4][8:].lower()
+
+    return None
+
+
+def _read_gpg_fp_from_file(key_path):
+    res = _gpg_show_keys(key_path)
+    return _parse_fp_from_gpg(res)
+
+
+def _expand_vars(path):
+    """
+    Expand variables like $releasever and $basearch to the target system version
+    """
+    r = path.replace("$releasever", get_target_major_version())
+    r = r.replace("$basearch", api.current_actor().configuration.architecture)
+    return r
+
+
+def _pubkeys_from_rpms(installed_rpms):
+    return [pkg.version for pkg in installed_rpms.items if pkg.name == 'gpg-pubkey']
+
+
+def _the_nogpgcheck_option_used():
+    return config.get_env('LEAPP_NOGPGCHECK', False) == '1'
+
+
+@suppress_deprecation(TMPTargetRepositoriesFacts)
+def process(consumer):
+    """
+    Process the repositories and find missing signing keys
+
+    UsedTargetRepositories doesn't contain baseurl attribute. So gathering
+    them from model TMPTargetRepositoriesFacts.
+    """
+
+    # when the user decided to ignore gpg signatures on the packages, we can ignore these checks altogether
```

Comment:
Keeping the decision on you. But I would say that when the `--nogpgcheck` is used by user, they should understand already what it does mean otherwise I would not expect them to use it.